src/utils/hooks/api/loans/use-get-account-positions.tsx

```diff
@@ -0,0 +1,49 @@
+import { BorrowPosition, LendPosition } from '@interlay/interbtc-api';
+import { AccountId } from '@polkadot/types/interfaces';
+import { useErrorHandler } from 'react-error-boundary';
+import { useQueries } from 'react-query';
+
+interface AccountPositionsData {
+  lendPositions: Array<LendPosition> | undefined;
+  borrowPositions: Array<BorrowPosition> | undefined;
+}
+
+const getAccountLendPositions = (accountId: AccountId): Promise<Array<LendPosition>> => {
+  return window.bridge.loans.getLendPositionsOfAccount(accountId);
+};
+const getAccountBorrowPositions = (accountId: AccountId): Promise<Array<BorrowPosition>> => {
+  return window.bridge.loans.getBorrowPositionsOfAccount(accountId);
+};
+
+const useGetAccountPositions = (accountId: AccountId | undefined): AccountPositionsData => {
+  const [
+    { data: lendPositionsData, error: lendPositionsError },
+    { data: borrowPositionsData, error: borrowPositionsError }
+  ] = useQueries([
+    {
+      queryKey: ['lend-positions', accountId?.toString()],
+      queryFn: async () => accountId && (await getAccountLendPositions(accountId)),
+      enabled: !!accountId
+    },
+    {
+      queryKey: ['borrow-positions', accountId?.toString()],
+      queryFn: async () => accountId && (await getAccountBorrowPositions(accountId)),
+      enabled: !!accountId
+    }
+  ]);
+
+  useErrorHandler(lendPositionsError);
+  useErrorHandler(borrowPositionsError);
+
+  if (accountId === undefined) {
```

Comment:
Instead of this, you can pass a initial value for each item of the array in useQueries

Reply:
Nice suggestion, thanks!